### PR TITLE
Add exception information into telemetry

### PR DESCRIPTION
**Pull request by dingmeng-xue**

### CODE REVIEW

File: src/Common/MetricHelper.cs [hunk 1]
```diff
@@ -281,12 +286,80 @@ private void PopulatePropertiesFromQos(AzurePSQoSEvent qos, IDictionary<string,
...
+            {
+                eventProperties.Add("pebcak", (qos.Exception.Data[AzurePSErrorDataKeys.ErrorKindKey] == ErrorKind.UserError).ToString());
+            }
+
+            if (qos.Exception != null && populateException)
             {
                 eventProperties.Add("exception-type", qos.Exception.GetType().ToString());
+                if (qos.Exception is CloudException)
+                {
+                    eventProperties.Add("exception-httpcode", ((CloudException)qos.Exception).Response?.StatusCode.ToString());
```
[the remark this answers is not in the document]
Comment (dingmeng-xue) [reply src/Common/MetricHelper.cs]: changed

File: src/Common/MetricHelper.cs [hunk 2]
```diff
@@ -281,12 +286,80 @@ private void PopulatePropertiesFromQos(AzurePSQoSEvent qos, IDictionary<string,
...
+                    {
+                        eventProperties.Add("exception-httpcode", ((CloudException)innerException).Response?.StatusCode.ToString());
+                    }
+                    innerException = innerException.InnerException;
+                }
+                if (innerExceptions.Count > 0)
+                {
+                    eventProperties.Add("exception-inner", string.Join(";", innerExceptions.Select(e => e.GetType().ToString())));
+                }
+                if (exceptionTrackAcceptModuleList.Contains(qos.ModuleName) || exceptionTrackAcceptCmdletList.Contains(qos.CommandName))
```
[the remark this answers is not in the document]
Comment (dingmeng-xue) [reply src/Common/MetricHelper.cs]: Added

File: src/Common/MetricHelper.cs [hunk 3]
```diff
@@ -281,12 +286,80 @@ private void PopulatePropertiesFromQos(AzurePSQoSEvent qos, IDictionary<string,
...
+                    eventProperties.Add("exception-stack", stack);
+                }
+
+                if (qos.Exception.Data != null)
+                {
+                    if (qos.Exception.Data.Contains(AzurePSErrorDataKeys.HttpStatusCode))
+                    {
+                        eventProperties.Add("exception-httpcode", qos.Exception.Data[AzurePSErrorDataKeys.HttpStatusCode].ToString());
+                    }
+                    IList<string> exceptionData = new List<string>();
```
[the remark this answers is not in the document]
Comment (dingmeng-xue) [reply src/Common/MetricHelper.cs]: Added.

File: src/Common/MetricHelper.cs [hunk 4]
```diff
@@ -303,6 +379,31 @@ private void PopulatePropertiesFromQos(AzurePSQoSEvent qos, IDictionary<string,
             }
         }
 
+        private static string[] exceptionTrackAcceptModuleList = { "Az.Accounts", "Az.Compute", "Az.AKS", "Az.ContainerRegistry" };
+        private static string[] exceptionTrackAcceptCmdletList = { "Get-AzKeyVaultSecret", "Get-AzKeyVaultCert" };
+
+        private static string ConvertFrameToString(StackFrame frame)
+        {
+            string[] fullNameParts = frame?.GetMethod()?.DeclaringType?.FullName?.Split('.');
```
Comment (erich-wang) [inline src/Common/MetricHelper.cs:387]: It seems that DeclaringType.FullName doesn't contain method name, so method name may be missed?

Comment (dingmeng-xue) [reply src/Common/MetricHelper.cs:387]: Great.

**Review by erich-wang — APPROVED** [review]

**Pull request merged**